alma/controllers/hook/StateHookController.php

```diff
@@ -43,38 +44,86 @@ final class StateHookController extends AdminHookController
     use OrderDataTrait;
 
     /**
+     * Execute query action on change state
      * @param $params
      *
      * @throws PrestaShopDatabaseException
      * @throws PrestaShopException
      */
     public function run($params)
     {
-        if (!Settings::isRefundEnabledByState()) {
-            return;
-        }
-
         $order = new Order($params['id_order']);
         $newStatus = $params['newOrderStatus'];
         if ($order->module !== 'alma' || !$order_payment = $this->getOrderPaymentOrFail($order)) {
             return;
         }
-
+        $alma = ClientHelper::defaultInstance();
+        if (!$alma) {
+            return;
+        }
         $id_payment = $order_payment->transaction_id;
-        if ($newStatus->id == Settings::getRefundState()) {
-            $alma = ClientHelper::defaultInstance();
-            if (!$alma) {
-                return;
+
+        try {
+            switch ($newStatus->id) {
+                case Settings::getRefundState():
+                    if (!Settings::isRefundEnabledByState()) {
+                        throw new NotPermittedBySettingsException('Refund option is not enabled');
+                    }
+                    $this->queryRefund($alma, $id_payment, $order);
+                    break;
+                case Settings::getPaymentTriggerState():
+                    if (!Settings::isPaymentTriggerEnabledByState()) {
+                        throw new NotPermittedBySettingsException('Payment trigger option is not enabled');
+                    }
+                    $this->queryTriggerPayment($alma, $id_payment, $order);
+                    break;
+                default :
+                    return;
             }
+        } catch (NotPermittedBySettingsException $e) {
+            Logger::instance()->info($e->getMessage());
+        }
+    }
+
+    /**
+     * Query Refund
+     *
+     * @param ClientHelper $alma
+     * @param string $id_payment
+     * @param Order $order
+     *
+     * @return void
+     */
+    private function queryRefund($alma, $id_payment, $order)
+    {
+        try {
+            $alma->payments->refund($id_payment, true);
+        } catch (RequestError $e) {
+            $msg = "[Alma] ERROR when creating refund for Order {$order->id}: {$e->getMessage()}";
+            Logger::instance()->error($msg);
 
-            try {
-                $alma->payments->refund($id_payment, true);
-            } catch (RequestError $e) {
-                $msg = "[Alma] ERROR when creating refund for Order {$order->id}: {$e->getMessage()}";
-                Logger::instance()->error($msg);
+            return;
+        }
+    }
 
-                return;
-            }
+    /**
+     * Query Trigger Payment
+     *
+     * @param ClientHelper $alma
+     * @param string $id_payment
+     * @param Order $order
+     *
+     * @return void
+     */
+    private function queryTriggerPayment($alma, $id_payment, $order)
```

Comment:
you can name this action just `triggerPayment`

Reply:
Done